src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -489,18 +494,18 @@ internal TypeWithAnnotations SubstituteTypeCore(AbstractTypeMap typeMap)
             {
                 newAnnotation = NullableAnnotation;
             }
-            else if (NullableAnnotation != NullableAnnotation.Oblivious)
+            else if (NullableAnnotation.IsNotAnnotated())
```

Comment:
>else if (NullableAnnotation.IsNotAnnotated()) [](start = 12, length = 45)

Would it make sense to add a check if old and new annotations are equal, right before this condition?